### PR TITLE
Stdout stderr to vec plus error exit codes

### PR DESCRIPTION
**Pull request by kdesjard**
For your consideration as this pull request also contains breaking changes.

This request allows for the capture of stdout and stderr regardless of whether a task completes successfully.  It also allows for the capturing of the OS's exit code in the case of a failure.

Thanks, Kris.

### CODE REVIEW

File: src/task/state.rs [hunk 1]
```diff
@@ -165,14 +166,14 @@ impl Output {
     }
 
     /// Construct an [`Output`]` with an error message.
-    pub fn error(msg: String) -> Self {
-        Self::Err(msg)
+    pub fn error(code: Option<i32>, msg: String) -> Self {
```
Comment (aminya) [inline src/task/state.rs:169]: @kdesjard This PR does not make sense for normal outputs when someone is not running Shell commands. Rust has rich error-handling features, which are much better than exit codes.

Also, this method is changed, which is a breaking change for all the users of Dagrs. We have to revert this change adding a new status field like `ErrWithExitCode`

**Comment by kdesjard** [comment]
Sorry about this and thanks for the updates/fixes.

**Comment by aminya** [comment]
> Sorry about this and thanks for the updates/fixes.

You're welcome!
